extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcTenantConfig.java

```diff
@@ -171,6 +171,20 @@ public void setTenantId(String tenantId) {
     @ConfigGroup
     public static class Credentials {
 
+        public static enum Type {
```

Comment:
I'm not sure about having this as a credential type. For me, credentials types as jwt, secret, mtls, etc. Which are related to the method that will be used to authenticate the client.

What if you have `quarkus.oidc.credentials.secret.method` instead ? Where `secret` would become a config object ?

Reply:
Hi @pedroigor This looks a bit complex ? In the simple case the user would have to do `quarkus.oidc.credentials.secret.value=secret` ?
I agree `quarkus.oidc.credentials.secret` should've ideally be part of some other object like client authentication, but it is already there now, so I though we could avoid another migration problem and qualify how that secret should be used, with this `Type` covering `client_secret_basic`, `client_secret_post` and easily `client_secret_jwt`.
When we will need to support `private_key_jwt` or mtls, then we'd have some new convention/holder for those and can just doc that `secret` property is ignored unless it is `client_secret_basic`, `client_secret_post` and easily `client_secret_jwt`.

Perhaps I can rename `Type` to `Method` and doc that this `Method` identifies how the credentials like 'secret' (or private_key or cert in the future) are used.

Would it be a reasonable compromise ?

Thanks

Reply:
@pedroigor To summarize, adding an extra `credentials.method` property will let us support 3 out of 4 client authentication methods defined in the OIDC core as they are all built around the use of the client secret, all without introducing a migration requirement. 
To support 2 more methods (private key for signing the jwt, mtls) we can either breaking configuration change and have a configuration object per every auth method or 3 at least, or add a Map of Maps to `credentials` and say the key must match the Method enum value, something like that, but we should probably deal with it later :-)